components/document-renderer.tsx

```diff
@@ -8,16 +9,30 @@ import { PageRenderer } from "./page-renderer";
  * Each of these cases map to a `section` found in the `.tina/settings.yml`
  *
  */
-export const DocumentRenderer = (props: Tina.SectionDocumentUnion) => {
-  switch (props.__typename) {
-    case "Pages_Document":
-      return <PageRenderer {...props.data} />;
+
+export const DocumentRenderer = ({
+  section,
+  document,
+}: {
+  section: String;
+  document: Tina.SectionDocumentUnion;
+}) => {
+  switch (section) {
+    case "shelves":
+      return <ShelvesRenderer {...document.data} />;
+    case "items":
+      return <ItemsRenderer {...document.data} />;
```

Comment:
I noticed that one point of confusion for me was the crosswalk between a `section` (or `collection`) and the GraphQL `__typename`.  In the test, I did not immediately put together that the `<DocumentRenderer />` is expecting to render something for each `section`.

To clear up this confusion, I changed the code to `switch` on `section` rather than `__typename`.  This led to passing `section` as an additional prop to `DocumentRenderer`, but it made understanding that the `section` I defined in the `schema.ts` is tied to the `DocumentRenderer`.

There might be a bigger discussion on the liberal use of `section` (which is leaving), `collection`, and `__typename` and how they're all interrelated.

Reply:
Also, I got TS errors for both `ShelvesRenderer` and `ItemsRenderer` that I was not getting.  I'm guessing it has to do with how I'm spreading `...document.data`?  Wouldn't mind figuring that out.

Reply:
When you switch on "section" how would Typescript know that document.data is a "Shelves" document vs an "Items" one? That's one affordance we get with __typename, since these types are all part of a union Typescript knows that a switch on __typename will result in only that type being passed through. So I'm guessing your error was related to that issue. 

On a slightly related note - I will say these types are only useful to an extent, but can possibly become cumbersome than they're worth. What you actually want is _query_ types, which would have an awareness of whether or not a field was included in the query.